opentargets_pharmgkb/ontology_apis.py

```diff
@@ -0,0 +1,63 @@
+import logging
+import os
+from functools import lru_cache
+
+import requests
+from cmat.trait_mapping.main import process_trait
+from cmat.trait_mapping.trait import Trait
+from requests import RequestException
+from retry import retry
+
+logging.basicConfig()
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+HOST = 'https://www.ebi.ac.uk'
+OLS_API_ROOT = f'{HOST}/ols/api'
+
+
+@lru_cache
+@retry(exceptions=(ConnectionError, RequestException), tries=4, delay=2, backoff=1.2, jitter=(1, 3))
+def get_chebi_iri(drug_name):
+    chebi_search_url = os.path.join(OLS_API_ROOT, f'search?ontology=chebi&q={drug_name}')
+    response = requests.get(chebi_search_url)
+    response.raise_for_status()
+    data = response.json()
+    if 'response' in data:
+        results = data['response']['docs']
+        candidates = set()
+        for result in results:
+            # Check that we've found the drug exactly (strict case-insensitive string match)
+            if result['label'].lower() == drug_name.lower():
+                candidates.add(result['iri'])
+        # Only return a result if we can find it unambiguously
+        if len(candidates) == 1:
+            return candidates.pop()
+    logger.warning(f'Could not find a CHEBI IRI for {drug_name}')
+    return None
+
+
+@lru_cache
+@retry(exceptions=(ConnectionError, RequestException), tries=4, delay=2, backoff=1.2, jitter=(1, 3))
+def get_efo_iri(phenotype_name):
+    if not phenotype_name:
+        return None
+
+    # Trait to store Zooma/OxO results - other attributes not used
+    trait = Trait(phenotype_name, None, None)
+    # Defaults from CMAT
+    filters = {'ontologies': 'efo,ordo,hp,mondo',
+               'required': 'cttv,eva-clinvar,clinvar-xrefs,gwas',
+               'preferred': 'eva-clinvar,cttv,gwas,clinvar-xrefs'}
```

Comment:
Should we have those be more explicitly defined rather than buried in the code ? 
I can see ourselves exploring the effect of the `required`/`preferred` parameters

Reply:
Good point, I was thinking the same about the required/preferred sources... extracted to variables at the top of the file.